src/vm/wren_core.wren

```diff
@@ -302,6 +302,52 @@ class List is Sequence {
     }
     return result
   }
+
+  sort() { sort_(this.map {|n| [n, n] }.toList) }
+
+  sort(key) { sort_(this.map {|n| [n, key.call(n)] }.toList) }
```

Comment:
I think another thing we want to consider here is do we want `sort` to change the existing object or return a new one. There's arguments for both.

Maybe have two functions: `sort` and `sorted`, where `sort` sorts in place and `sorted` returns a sorted copy.